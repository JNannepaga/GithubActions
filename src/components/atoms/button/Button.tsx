import { Button as AntDesignButton} from 'antd';
import { ButtonProps as AntDesignButtonProps} from 'antd/lib/button';
import React from 'react';
import { Location } from 'history';
import { Link as RouterLink } from 'react-router-dom';

export interface ButtonProps extends AntDesignButtonProps{
    to?: Location;
    children?: React.ReactNode
}

export const Button: React.FunctionComponent<ButtonProps> = ({
    to,
    type,
    children,
    ...props
}) =>{
    const button = <AntDesignButton type={type} {...props}>
                        {children}
                    </AntDesignButton>;
    return to ? (<RouterLink to={to}>{button}</RouterLink>) : (button)
} 