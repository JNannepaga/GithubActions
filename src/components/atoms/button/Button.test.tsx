import React from "react";
import {Button} from "./Button";
import renderer from "react-test-renderer";

describe("Components",()=>{
    describe("Atoms",()=>{
        describe("Button",()=>{
            it("matches snapshot",()=>{
               const wrapper = renderer.create(<Button/>).toJSON();
               expect(wrapper).toMatchSnapshot();
            })
        })
    })
})