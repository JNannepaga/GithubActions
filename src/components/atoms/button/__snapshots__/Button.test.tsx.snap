// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Atoms Button matches snapshot 1`] = `
<button
  className="ant-btn ant-btn-default"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;
